src/services/ServiceLocator.hh

```diff
@@ -58,9 +58,9 @@ public:
     /// pointer and delete it at the end of execution or when a different implementation is provided.
     /// </param>
     template <class TClass>
-    static void Provide(TClass* pInstance) noexcept
+    static void Provide(std::unique_ptr<TClass> pInstance) noexcept
     {
-        Service<TClass>::s_pInstance.reset(pInstance);
+        Service<TClass>::s_pInstance = std::move(pInstance);
```

Comment:
I just made it like this by default. If `s_pInstance `is a `unique_ptr`, it makes sense that `pInstance` is also a `unique_ptr`. As far as this method is concerned it's leak proof with no "`new`"s anywhere. No copying was done at all.